.github/workflows/ci.yml

```
@@ -0,0 +1,240 @@
+on: [push, pull_request]
+name: Continuous Integration
+
+jobs:
+  check-if-tag:
+    name: Get Git Tag
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{steps.deployment.outputs.version}}
+    steps:
+      - name: Set Variables
+        id: deployment
+        shell: bash
+        run: |
+          if [ $(echo ${{github.ref}} | grep -q "refs/tags/"; echo $?) == 0 ]; then
+            echo "::set-output name=VERSION::${GITHUB_REF#/refs\/tags\//}"
+          else
+            echo "::set-output name=VERSION::0.0.0+${{github.run_id}}"
+          fi
+  test:
+    name: Test
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install .NET 5.0.x
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: "5.0.x"
+
+      - name: Compile
+        run: dotnet build -c Debug build/Desktop.proj
+
+      - name: Test
+        run: dotnet test $pwd/*.Tests/bin/Debug/*/*.Tests.dll --settings $pwd/build/vstestconfig.runsettings --logger:nunit
+        shell: powershell
+
+      - name: Report NUnit Logs
+        uses: MirrorNG/nunit-reporter@v1.0.9
+        if: ${{ always() }}
+        with:
+          path: "./TestResults/TestResults.xml"
+          access-token: ${{ secrets.GITHUB_TOKEN }}
+
+  inspect-code:
+    name: Code Quality
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # FIXME: Tools won't run in .NET 5.0 unless you install 3.1.x LTS side by side.
+      # https://itnext.io/how-to-support-multiple-net-sdks-in-github-actions-workflows-b988daa884e
+      - name: Install .NET 3.1.x LTS
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: "3.1.x"
+
+      - name: Install .NET 5.0.x
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: "5.0.x"
+
+      - name: Restore Tools
+        run: dotnet tool restore
+
+      - name: Restore Packages
+        run: dotnet restore
+
+      - name: CodeFileSanity
+        run: |
+          # TODO: Add ignore filters and GitHub Workflow Command Reporting in CFS. That way we don't have to do this workaround.
+          # FIXME: Suppress warnings from templates project
+          # https://unix.stackexchange.com/questions/104641/how-can-i-suppress-output-from-grep-so-that-it-only-returns-the-exit-status
+          dotnet codefilesanity | while read -r line; do
+             if [ $(echo $line| grep -q "./osu.Framework.Templates/"; echo $?) == 1 ]; then
```

```suggestion
             if [[ "$line" != *"/osu.Framework.Templates/"* ]]; then
```

No need to create a process each line.

---

Not really a fan of Bash-conditionals, as much as possible, I use POSIX.

---

Personally I think the proposal reads better.

---

tested it and it works just as well, thanks for the suggestion. Though I'd prefer this would be using POSIX conditionals for consistency but this will do for now.